local-cli/eject/eject.js

```diff
@@ -64,6 +70,10 @@ function eject() {
     process.exit(1);
   }
 
+  const iconConfig = (typeof appConfig.icon === 'string' ? {default: appConfig.icon.default} : appConfig.icon  ) || {};
```

Comment:
I believe this should be changed to:

```
const iconConfig = (typeof appConfig.icon === 'string' ? {default: appConfig.icon} : appConfig.icon  ) || {};
```

For each PR to react native, we also expect to see a test plan, that would cover the edge cases of the changes you're making. This is one edge case that I would expect to be in the test plan, and hopefully it would help surface the issue sooner